.github/workflows/release.yml

```diff
@@ -0,0 +1,98 @@
+name: goreleaser
+
+on:
+  pull_request:
+  push:
+    tags:
+      - 'v*'
+
+jobs:
+  goreleaser:
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      -
+        name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.16.0
+      -
+        name: Run GoReleaser
+        uses: goreleaser/goreleaser-action@v2
+        if: startsWith(github.ref, 'refs/tags/')
+        with:
+          version: 0.162.0
+          args: release --rm-dist --debug
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      -
+        uses: actions/github-script@v4
+        id: get-checksums-from-draft-release
+        if: startsWith(github.ref, 'refs/tags/')
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          result-encoding: string
+          script: |
+            var crypto = require('crypto');
+            const { owner, repo } = context.repo;
+
+            // https://docs.github.com/en/rest/reference/repos#list-releases
+            // https://octokit.github.io/rest.js/v18#repos-list-releases
+            var releases = await github.repos.listReleases({
+              owner: owner,
+              repo: repo
+            });
+
+            var crypto = require('crypto')
+            var fs = require('fs')
+            const url = require('url');
+            const https = require('https');
+
+            checksums = {}
+            for (const r of releases["data"]) {
```

Comment:
hmmm, this part of the release GH action is to verify that the binaries uploaded by goreleaser match the same sha as binaries generated via the `hack/build-binaries.sh`

The concern is that `goreleaser` *could* have been compromised, and thus, uploaded a compromised binary. 

So, compare the sha's of the uploaded binary built with `goreleaser`, with the sha's generated using native `go build` via `hack/build-binaries.sh`

> upload succeeded , nothing was tampered

Perhaps I should just assume nothing was tampered. Although leaving this 'verification' step is hopefully one less attack vector.

Reply:
> So, compare the sha's of the uploaded binary built with goreleaser, with the sha's generated using native go build via hack/build-binaries.sh

oh very interesting. i think i missed that part. are we able to reliably guarantee that our script is equivalent to whatever goreleaser is producing?

Reply:
Yes, locally and when run as a GH action it produces the same `sha` as our internal `hack/build.sh` script.

also, from the debug logs goreleaser provides, it runs the equivalent `go build` command...

However, `goreleaser` is a binary. And while today it generates the same `sha` it might not tomorrow (assuming it is compromised / malicious)

Reply:
Had a call with dk. Marking this conversation as resolved.